metro.go

```diff
@@ -93,16 +106,19 @@ func initLogging(to_file bool, level string) {
 	logger, err := log.LoggerFromConfigAsBytes(logConfig)
 	if err != nil {
 		log.Criticalf("Unable to initiate logger: %s", err)
-		os.Exit(1)
+		panic(Exit{1})
 	}
 	log.ReplaceLogger(logger)
+	return logger
 
 }
 
 func main() {
+	defer handleExit()
 	flag.Parse()
 
-	initLogging(true, "warning")
+	logger := initLogging(true, "warning")
+	defer logger.Flush()
```

Comment:
I did it differently because we replaced the logger, and I thought I had to make the call on a per logger basis.